autosklearn/pipeline/components/data_preprocessing/one_hot_encoding/minority_coalescer.py

```diff
@@ -0,0 +1,73 @@
+import numpy as np
+import scipy.sparse
+
+import autosklearn.pipeline.implementations.MinorityCoalescer
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+from ConfigSpace.hyperparameters import CategoricalHyperparameter, \
+    UniformFloatHyperparameter
+from ConfigSpace.conditions import EqualsCondition
+
+from autosklearn.pipeline.components.base import AutoSklearnPreprocessingAlgorithm
+from autosklearn.pipeline.constants import *
+from autosklearn.util.common import check_for_bool
+
+
+class MinorityCoalescer(AutoSklearnPreprocessingAlgorithm):
+    """ Group together categories which occurence is less than a specified minimum fraction.
+    """
+
+    def __init__(self, use_minimum_fraction=True, minimum_fraction=0.01, random_state=None):
+        self.use_minimum_fraction = use_minimum_fraction
+        self.minimum_fraction = minimum_fraction
+        
+
+    def _fit(self, X, y=None):
+        self.use_minimum_fraction = check_for_bool(self.use_minimum_fraction)
+        if self.use_minimum_fraction is False:
+            self.minimum_fraction = None
+        else:
+            self.minimum_fraction = float(self.minimum_fraction)
+
+        self.preprocessor = autosklearn.pipeline.implementations.MinorityCoalescer\
+            .MinorityCoalescer(minimum_fraction=self.minimum_fraction)
+
+        return self.preprocessor.fit_transform(X)
+
+    def fit(self, X, y=None):
+        self._fit(X, y)
+        return self
+
+    def fit_transform(self, X, y=None):
+        return self._fit(X)
+
+    def transform(self, X):
+        if self.preprocessor is None:
+            raise NotImplementedError()
+        return self.preprocessor.transform(X)
+
+    @staticmethod
+    def get_properties(dataset_properties=None):
+        return {'shortname': 'coalescer',
+                'name': 'Categorical minority coalescer',
+                'handles_regression': True,
+                'handles_classification': True,
+                'handles_multiclass': True,
+                'handles_multilabel': True,
+                # TODO find out of this is right!
+                'handles_sparse': True,
+                'handles_dense': True,
+                'input': (DENSE, SPARSE, UNSIGNED_DATA),
+                'output': (INPUT,),}
+
+    @staticmethod
+    def get_hyperparameter_search_space(dataset_properties=None):
+        cs = ConfigurationSpace()
+        use_minimum_fraction = CategoricalHyperparameter(
```

Comment:
This hyperparameter is now really weird. I believe we should do this similar to the one hot encoder, where there's basically an alternative component which does nothing.

Reply:
Ok